superset/connectors/sqla/models.py

```diff
@@ -821,15 +816,21 @@ def query(self, query_obj):
         status = utils.QueryStatus.SUCCESS
         error_message = None
         df = None
-        db_engine_spec = self.database.db_engine_spec
         try:
             df = self.database.get_df(sql, self.schema)
-            if self.mutated_labels:
-                df = df.rename(index=str, columns=self.mutated_labels)
-            db_engine_spec.mutate_df_columns(df, sql, query_str_ext.labels_expected)
+            labels_expected = query_str_ext.labels_expected
```

Comment:
So as per https://github.com/apache/incubator-superset/pull/6831, I would be pushing this mutation inside the get_df (by giving the get_df call a functor).

Reply:
Yeah I didn't do any of those changes here, I wanted to limit the scope of this PR to just the cleaning up bit. Let's tackle that in your original PR once we get consensus on this one first.